src/Data/Either/Open.purs

```diff
@@ -0,0 +1,146 @@
+
+in4 :: forall z a b c d. a -> Nested.Either5 z a b c d
+in4 v = Left (Left (Left (Right v)))
+
+in5 :: forall z a b c d e. a -> Nested.Either6 z a b c d e
+in5 v = Left (Left (Left (Left (Right v))))
+
+in6 :: forall z a b c d e f. a -> Nested.Either7 z a b c d e f
+in6 v = Left (Left (Left (Left (Left (Right v)))))
+
+in7 :: forall z a b c d e f g. a -> Nested.Either8 z a b c d e f g
+in7 v = Left (Left (Left (Left (Left (Left (Right v))))))
+
+in8 :: forall z a b c d e f g h. a -> Nested.Either9 z a b c d e f g h
+in8 v = Left (Left (Left (Left (Left (Left (Left (Right v)))))))
+
+in9 :: forall z a b c d e f g h i. a -> Nested.Either10 z a b c d e f g h i
+in9 v = Left (Left (Left (Left (Left (Left (Left (Left (Right v))))))))
+
+in10 :: forall z a b c d e f g h i j. a -> Nested.Either10 (Either z a) b c d e f g h i j
+in10 v = Left (Left (Left (Left (Left (Left (Left (Left (Left (Right v)))))))))
+
+at1 :: forall z r a. r -> (a -> r) -> Nested.Either2 z a -> r
+at1 b a y = case y of
+  Right r -> a r
+  _ -> b
+
+at2 :: forall z r a b. r -> (a -> r) -> Nested.Either3 z a b -> r
+at2 b a y = case y of
+  Left (Right r) -> a r
+  _ -> b
+
+at3 :: forall z r a b c. r -> (a -> r) -> Nested.Either4 z a b c -> r
+at3 b a y = case y of
+  Left (Left (Right r)) -> a r
+  _ -> b
+
+at4 :: forall z r a b c d. r -> (a -> r) -> Nested.Either5 z a b c d -> r
+at4 b a y = case y of
+  Left (Left (Left (Right r))) -> a r
+  _ -> b
+
+at5 :: forall z r a b c d e. r -> (a -> r) -> Nested.Either6 z a b c d e -> r
+at5 b a y = case y of
+  Left (Left (Left (Left (Right r)))) -> a r
+  _ -> b
+
+at6 :: forall z r a b c d e f. r -> (a -> r) -> Nested.Either7 z a b c d e f -> r
+at6 b a y = case y of
+  Left (Left (Left (Left (Left (Right r))))) -> a r
+  _ -> b
+
+at7 :: forall z r a b c d e f g. r -> (a -> r) -> Nested.Either8 z a b c d e f g -> r
+at7 b a y = case y of
+  Left (Left (Left (Left (Left (Left (Right r)))))) -> a r
+  _ -> b
+
+at8 :: forall z r a b c d e f g h. r -> (a -> r) -> Nested.Either9 z a b c d e f g h -> r
+at8 b a y = case y of
+  Left (Left (Left (Left (Left (Left (Left (Right r))))))) -> a r
+  _ -> b
+
+at9 :: forall z r a b c d e f g h i. r -> (a -> r) -> Nested.Either10 z a b c d e f g h i -> r
+at9 b a y = case y of
+  Left (Left (Left (Left (Left (Left (Left (Left (Right r)))))))) -> a r
+  _ -> b
+
+at10 :: forall z r a b c d e f g h i j. r -> (a -> r) -> Nested.Either10 (Either z a) b c d e f g h i j -> r
+at10 b a y = case y of
+  Left (Left (Left (Left (Left (Left (Left (Left (Left (Right r))))))))) -> a r
+  _ -> b
+
+runEither1 :: forall a. Either1 a -> a
+runEither1 y = unsafePartial case y of
+  Right r -> r
+
+either2 :: forall r a b. (a -> r) -> (b -> r) -> Either2 a b -> r
+either2 a b y = case y of
+  Right r -> b r
+  Left _1 -> a (runEither1 _1)
+
+either3 :: forall r a b c. (a -> r) -> (b -> r) -> (c -> r) -> Either3 a b c -> r
+either3 a b c y = case y of
+  Right r -> c r
+  Left _1 -> either2 a b _1
```

Comment:
All the delegation improves code re-use, but decreases performance. Though the argument can be made if you want performance, you should not be using nested eithers. Nonetheless, if you do use nested eithers, you'd probably prefer @natefaubion made them as fast as possible. 😄